syntax = "proto3";

import "basic.proto";
import "common.proto";
import "perception_base.proto";

package autodrive.workflow.proto;

enum MapObjectType {
  kMapObjectTypeUnknown = 0;
  kMapObjectTypeTrafficSign = 1;
  kMapObjectTypeTrafficLight = 2;
  kMapObjectTypeTrafficLightBulb = 3;
  kMapObjectTypeLaneMarking = 4;
  kMapObjectTypeStopLine = 5;
  kMapObjectTypeSpeedBump = 6;
  kMapObjectTypePole = 7;
  kMapObjectTypeCrossWalk = 8;
  kMapObjectTypeZone = 9;
  kMapObjectTypeParkingSlot = 10;
  kMapObjectTypeTrafficCone = 11;
  kMapObjectTypeParkingLockClose = 12;
  kMapObjectTypeParkingLockOpen = 13;
  kMapObjectTypeParkingColumn = 14;
  kMapObjectTypeParkingAFrameSign = 15;
  kMapObjectTypeJunction = 16;
  kMapObjectTypeParkingLock = 17;
  /// @brief Connected to the wall
  kMapObjectTypeCementColumn_Yes = 18;
  /// @brief Not connected to the wall
  kMapObjectTypeCementColumn_No = 19;
}

message MapObjectAttr {
  int32 shape = 1;
  float cycle_time = 2;
  int32 traversable = 3;
  float height = 4;
  float radius = 5;
  int32 status = 6;
  string value = 7;
  bool dynamic = 8;
  perception_base.ImageSpace img_info = 9;
}

enum MapObjectSubType {
  kParkLockTypeUnknown = 0;
  kParkLockTypeOpen = 1;
  kParkLockTypeClose = 2;
  kFireHydrantTypeAir = 3;
  kFireHydrantTypeGnd = 4;
}

message MapObject {
  int32 id = 1;
  int32 cam_id = 2;
  int64 time_stamp = 3;
  int32 conf = 4;
  MapObjectType type = 5;
  MapObjectSubType sub_type = 6;
  perception_base.Polygon border = 7;
  int32 life_time = 8;
  int32 age = 9;
  string str_id = 10;
  MapObjectAttr attr = 11;
  Vector3f position = 12;
}

message MapObjectsMsg {
  // Header
  MsgHeader header = 1;
  // Error
  Error error = 2;
  float conf_scale = 3;
  repeated MapObject objects = 4;
}